### PR TITLE
Replace multiple select() overloads with variadic template (Issue #1907)

### PR DESCRIPTION
**Pull request by steven-johnson**

### CODE REVIEW

File: src/IROperator.h [hunk 1]
```diff
@@ -861,134 +861,10 @@ inline Expr select(Expr condition, Expr true_value, Expr false_value) {
...
-                  c3, v3,
-                  c4, v4,
-                  c5, v5,
-                  c6, v6,
-                  c7, v7,
-                  c8, v8,
-                  c9, v9,
-                  select(c10, v10, default_val));
+template<typename... Args>
+inline Expr select(Expr c0, Expr v0, Expr c1, Expr v1, Args&&... args) {
```
Comment (abadams) [inline src/IROperator.h]: The first four args can be const-ref

**Comment by dsharletg** [comment]
Should we also have select(Expr, Expr, int), select(Expr, int, Expr), select(Expr, Expr, float), select(Expr, float, Expr) overloads, similar to min/max? That way, if someone does select(c, u8_value, 2), they get a reasonable result. This was intractable before with the overload approach to implementing select for multiple conditions, but now it seems reasonable to support.

**Comment by steven-johnson** [comment]
The overloads you describe should come for free with the variadic overloads, since they all resolve to one of the existing pairwise overloads.

**Comment by dsharletg** [comment]
Hmm, not sure I followed that, but looking closer, I see that the implemention of select(Expr, Expr, Expr) has some type coercion logic, implemented differently from the way min/max does it. I need to go back and look at my code that had a problem with this to see why it wasn't working.

**Comment by steven-johnson** [comment]
It's no wonder you didn't follow it, since I just realized I was talking about the similar min/max PR instead of this one :-)

**Comment by abadams** [comment]
Looks like the fft app needs fixing up: https://buildbot.halide-lang.org/master/builders/linux-64-gcc49-391/builds/167/steps/make%20test_apps/logs/stdio

**Pull request merged**